benchopt/plotting/plot_objective_curve.py

```diff
@@ -176,39 +145,16 @@ def compute_quantiles(df_filtered):
     return q1, q9
 
 
-def get_solver_color(solver):
-    if solver in html_solver_styles and 'color' in html_solver_styles[solver]:
-        return html_solver_styles[solver]['color']
-
-    idx = len(html_solver_styles)
-    color = CMAP(idx % CMAP.N)
-    color = tuple(255*x if i != 3 else x for i, x in enumerate(color))
-    color = f'rgba{color}'
+def get_solver_style(solver, plotly=True):
+    idx = solvers_idx.get(solver, len(solvers_idx))
+    solvers_idx[solver] = idx
 
-    if solver in html_solver_styles:
-        html_solver_styles[solver]['color'] = color
-    else:
-        html_solver_styles[solver] = {
-            'color': color
-        }
+    color = colors[idx % len(colors)]
+    marker = markers[idx % len(markers)]
 
-    return color
-
-
-def get_solver_marker(solver):
-    if solver in html_solver_styles and 'marker' in html_solver_styles[solver]:
-        return html_solver_styles[solver]['marker']
-
-    markers = {i: i for i, v in enumerate(plt.Line2D.markers)}
-
-    i = len(html_solver_styles)
-    marker = markers[i % len(markers)]
-
-    if solver in html_solver_styles:
-        html_solver_styles[solver]['marker'] = marker
-    else:
-        html_solver_styles[solver] = {
-            'marker': marker
-        }
+    if plotly:
+        color = tuple(255*x if i != 3 else x for i, x in enumerate(color))
+        color = f'rgba{color}'
+        marker = list(markers.values()).index(marker)
```

Comment:
I think we can imply do:
```suggestion
        marker = idx
```